developer/system/template-manager.md

```diff
@@ -0,0 +1,241 @@
+In turn, this creates inconveniences and confusion for users ([#1403], [#4518]).
+
+Also, the usage of RPM packages meant that installing a template results in arbitrary code execution, which is not ideal.
+
+Besides distribution, users may also wish to have an integrated template management application ([#2062], [#2064], [#2534], [#3040]), as opposed to the situation where multiple programs are required for different purposes, e.g., `qubes-dom0-update`, `dnf`, `qvm-remove`, `qubes-manager`.
+
+To tackle these issues, `qvm-template` is created.
+It strives to provide not only a better mechanism for handling template installation but also a consistent user-facing interface to deal with template management.
+
+[#996]: https://github.com/QubesOS/qubes-issues/issues/996
+[#1647]: https://github.com/QubesOS/qubes-issues/issues/1647
+[#2061]: https://github.com/QubesOS/qubes-issues/issues/2061
+[#839]: https://github.com/QubesOS/qubes-issues/issues/839
+[#5509]: https://github.com/QubesOS/qubes-issues/issues/5509
+[#1385]: https://github.com/QubesOS/qubes-issues/issues/1385
+[#1453]: https://github.com/QubesOS/qubes-issues/issues/1453
+[#1403]: https://github.com/QubesOS/qubes-issues/issues/1403
+[#4518]: https://github.com/QubesOS/qubes-issues/issues/4518
+[#2062]: https://github.com/QubesOS/qubes-issues/issues/2062
+[#2064]: https://github.com/QubesOS/qubes-issues/issues/2064
+[#2534]: https://github.com/QubesOS/qubes-issues/issues/2534
+[#3040]: https://github.com/QubesOS/qubes-issues/issues/3040
+[discussion thread 1]: https://groups.google.com/forum/#!topic/qubes-devel/rwc2_miCNNE/discussion
+[discussion thread 2]: https://groups.google.com/forum/#!topic/qubes-users/uQEUpv4THsY/discussion
+
+## Features
+
+- Install/reinstall/downgrade/upgrade templates, either from local packages or remote repositories
+  - Ability to install templates in alternative pools
+  - Possibility for the template package to specify options such as the kernel or `virt_mode` used
+- List and show information about local and available templates
+  - Machine-readable output for easy extension
+- Search for templates
+- Remove templates
+  - Ability to disassociate or remove VMs based on the template to be removed
+- Show available repositories
+- Works in both dom0 and management VMs by utilizing the Admin API
+- Works well with existing tools
+- Command-line interface with DNF-like usage
+- A graphical interface also available
+
+## Package Format
+
+The RPM package format is still used.
+However, the contents are manually extracted instead of installing the whole package.
+This allows us to take advantage of existing tools for things like repository management.
+We can also avoid the burden of dealing with verification, reducing the risk of issues like [QSB-28].
+
+The package name should be in the form `qubes-template-${template_name}`.
+
+For the package metadata (e.g., summary, description), one needs to ensure that the character `|` does not appear.
```

Comment:
I'd rewrite this for clarity, something like 'In the package metadata (summary, description etc.) should not contain the `|` character to avoid various errors (it is used as an internal separator).'